src/adapter/src/coord/sequencer.rs

```diff
@@ -1032,10 +1032,24 @@ impl<S: Append + 'static> Coordinator<S> {
             item: CatalogItem::Sink(catalog_sink.clone()),
         }];
 
+        let from_name = self
+            .catalog
+            .get_entry(&catalog_sink.from)
+            .name()
+            .item
+            .clone();
         let result = self
             .catalog_transact(Some(&session), ops, move |txn| {
                 // Validate that the from collection is in fact a persist collection we can export.
-                txn.dataflow_client.storage().collection(sink.from)?;
+                txn.dataflow_client
+                    .storage()
+                    .collection(sink.from)
+                    .map_err(|e| match e {
+                        StorageError::IdentifierMissing(_) => AdapterError::Unstructured(anyhow!(
+                            "{from_name} is not a persistent collection, and cannot be exported as a sink"
```

Comment:
I'm not 100% sure of this wording... in particular, is a "persistent collection" a concept that makes sense to a user, or just something we talk about internally?

Reply:
Yeah, `persisted collection` is a little inside baseball. What about...

```rust
        let from = self.catalog.get_entry(&catalog_sink.from);
        let from_name = from.name().item.clone();
        let from_type = from.item().typ().to_string();
```

```rust
                            "{from_name} is a {from_type}, and cannot be exported as a sink"
```

Reply:
sounds good to me! change pushed.